File: src/main/java/info/kaara/it100/mqtt/Topic.java
package info.kaara.it100.mqtt;

import java.util.Arrays;

public enum Topic {

	STATE("home/alarm"),
	COMMAND("home/alarm/set");

	private final String topic;

	Topic(String topic) {
		this.topic = topic;
	}

	public String getTopic() {
		return topic;
	}

	public static Topic toTopic(String topic) {
		return Arrays.stream(values()).filter(topic1 -> topic1.topic.equalsIgnoreCase(topic)).findFirst().orElse(null);
	}


}

File: src/main/kotlin/info/kaara/it100/mqtt/alarm/Topic.kt
package info.kaara.it100.mqtt.alarm

enum class Topic(val topic: String) {

    STATE("home/alarm"),
    COMMAND("home/alarm/set");

    companion object {
        fun toTopic(topic: String): Topic {
            return values().filter { t -> t.topic.equals(topic) }.first()
        }
    }
}